homeassistant/components/stiebel_eltron/entity.py

```diff
@@ -0,0 +1,60 @@
+"""StiebelEltronEntity class."""
+
+from dataclasses import dataclass
+
+from pystiebeleltron import IsgRegisters, IsgRegistersNone
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ATTR_MANUFACTURER, DOMAIN
+from .coordinator import StiebelEltronDataCoordinator
+from .data import StiebelEltronConfigEntry
+
+
+@dataclass(frozen=True, kw_only=True)
+class StiebelEltronEntityDescription(EntityDescription):
+    """Entity description for stiebel eltron with modbus register."""
+
+    modbus_register: IsgRegisters
+
+
+class StiebelEltronEntity(CoordinatorEntity[StiebelEltronDataCoordinator]):
+    """STIEBEL ELTRON entity base class."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: StiebelEltronDataCoordinator,
+        config_entry: StiebelEltronConfigEntry,
+    ) -> None:
+        """Initialize the entity base class."""
+        super().__init__(coordinator)
+        self.config_entry = config_entry
+        self.modbus_register: IsgRegisters = IsgRegistersNone.NONE
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info of the entity."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.coordinator.name)},
+            configuration_url=f"http://{self.coordinator.host}",
+            name=self.coordinator.name,
+            model=self.coordinator.model,
+            manufacturer=ATTR_MANUFACTURER,
+        )
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added.
+
+        This only applies when first added to the entity registry.
+        """
+        return self.available
```

Comment:
I kinda disagree? We should only disable impopular or noisy entities. Combined with `available` it kinda looks like "if we don't have a register value (and assuming the modbus interface doesn't change on runtime) the entity is unavailable (and it never will) and then we just disable it". Instead, if an entity has no use or can't be used, we shouldn't create it in the first place